### DISCUSSION TITLE
rotor contribution

### DISCUSSION THREAD

**Discussion by memy009** [JSBSim-Team/jsbsim#934] [Q&A]
Hi,

Below a part of the modeling of the propulsion part for an F450. 

Can someone explain me what <p-factor/> stands for exacltly? 

According to my my researches, it says that it gives the contribution of each engine to the movement : like when the disposition of the propulsion system is symetric, the factor is equal to zero but when it's not, an aerodynamic study have to be done ... if it is the case, what kind of aerodynamic study? could you give me an exemple ?

Thank U. 

```xml
<propulsion>

  <engine file="DJI_E305" name="front right">

    <thruster file="DJI_9450">

      <location unit="M">

        <x>-0.1651</x>

        <y>0.1651</y>

        <z>0.025</z>

      </location>

      <orient unit="DEG">

        <roll>0</roll>

        <pitch>90</pitch>

        <yaw>0</yaw>

      </orient>

      <sense>1.0</sense>

      <p_factor>0.0</p_factor>

    </thruster>

  </engine>
```

**Comment by seanmcleod** [comment]
As mentioned below, the propellor produces thrust and 2 moments. JSBSim then uses this force and the moments and applies them to the aircraft.

https://github.com/JSBSim-Team/jsbsim/blob/6c1abbe2ba53268eaf2178bebd946af0e49c7d43/src/models/propulsion/FGPropeller.h#L116-L120

https://github.com/JSBSim-Team/jsbsim/blob/6c1abbe2ba53268eaf2178bebd946af0e49c7d43/src/models/propulsion/FGPropeller.h#L131-L143

https://github.com/JSBSim-Team/jsbsim/blob/6c1abbe2ba53268eaf2178bebd946af0e49c7d43/src/models/propulsion/FGPropeller.cpp#L258-L262

**Comment by agodemar** [comment]
The engine has a thruster. 
Example: a propeller. 
The engine puts the propeller at a given angular speed around its axis.
The rotating propeller (`<truster />`) develops a thrust force (that also depends on how fast the aircraft travels in the air, that is, on the aircraft state).
The thrust force is applied to a point (`<location />`) and has an orientation with respect to the airframe (`<orient />`).
This thrust pushes the airframe into the air and determines its movement.
The attribute `file="DJI_9450"` indicates another file containing the [propeller's thrust and couple coefficients](https://web.mit.edu/16.unified/www/FALL/thermodynamics/notes/node86.html), in the form of lookup tables. These are the result of aerodynamic studies, that typically provide the coefficients as functions of [propeller's advance ratio.](https://en.wikipedia.org/wiki/Advance_ratio) The state propagation algorithm within JSBSim uses the coefficients at runtime to calculate the thrust and the engine state.
There is actually [a lot of background behind](https://web.mit.edu/16.unified/www/FALL/thermodynamics/notes/node79.html) that.

**Comment by memy009** [comment]
Thank you for your responses .

**Reply by agodemar** [reply]
@memy009 I've updated my previous reply with some useful links to the theory of aircraft propulsion (also good for drones and rotorcraft)

**Reply by memy009** [reply]
I'll check it , thank you for your help :)
